x/nft-transfer/keeper/relay.go

```diff
@@ -0,0 +1,154 @@
+package keeper
+
+import (
+	"github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clienttypes "github.com/cosmos/ibc-go/v5/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v5/modules/core/24-host"
+	coretypes "github.com/cosmos/ibc-go/v5/modules/core/types"
+	"github.com/crypto-org-chain/chain-main/v4/x/nft-transfer/types"
+)
+
+// SendTransfer handles nft-transfer sending logic.
+// A sending chain may be acting as a source or sink zone.
+//
+// when a chain is sending tokens across a port and channel which are
+// not equal to the last prefixed port and channel pair, it is acting as a source zone.
+// when tokens are sent from a source zone, the destination port and
+// channel will be prefixed onto the classId (once the tokens are received)
+// adding another hop to the tokens record.
+//
+// when a chain is sending tokens across a port and channel which are
+// equal to the last prefixed port and channel pair, it is acting as a sink zone.
+// when tokens are sent from a sink zone, the last prefixed port and channel
+// pair on the classId is removed (once the tokens are received), undoing the last hop in the tokens record.
+//
+// For example, assume these steps of transfer occur:
+// A -> B -> C -> A -> C -> B -> A
+//
+// |                    sender  chain                      |                       receiver     chain              |
+// | :-----: | -------------------------: | :------------: | :------------: | -------------------------: | :-----: |
+// |  chain  |                    classID | (port,channel) | (port,channel) |                    classID |  chain  |
```

Comment:
is there any length limitation to the `classID`

Reply:
Different chains can have different limitations. In our case, here is the `denomID` validation function: https://github.com/crypto-org-chain/chain-main/blob/d86e35e05dd38738990a88099ccb1ed015425a14/x/nft/types/validation.go#L28